File: Area/app_mobile/lib/Model/Cards.dart


import 'package:json_annotation/json_annotation.dart';
import 'package:shared_preferences/shared_preferences.dart';

part 'Cards.g.dart';


@JsonSerializable()
class CardsList {
  @JsonKey(name: "data")
  List<Cards> list;


  CardsList(this.list);

  factory CardsList.fromJson(Map<String, dynamic> json) => _$CardListFromJson(json);

  Map<String, dynamic> toJson() => _$CardListToJson(this);
}

@JsonSerializable()
class Cards {
  @JsonKey(name: '')
  String name;
  String ServiceAction;
  String Action;
  String ActionInfo;
  String ServiceReaction;
  String Reaction;
  String ReactionInfo;

  Cards(
      this.name,
      this.ServiceAction,
      this.Action,
      this.ActionInfo,
      this.ServiceReaction,
      this.Reaction,
      this.ReactionInfo
      );

  factory Cards.fromJson(Map<String, dynamic> json) => _$CardsFromJson(json);

  Map<String, dynamic> toJson() => _$CardsToJson(this);
}

File: Area/web_app/src/Cards.js
import React, { Component } from "react";

import './style.css';

class Cards extends React.Component {
    constructor() {
        super();
        this.state = {
            email: "email",
            name: "name",
            serviceAction: "serviceAction",
            action: "action",
            actionInfo: "actionInfo",
            serviceReaction: "serviceReaction",
            reaction: "reaction",
            reactionInfo: "reactionInfo"
        };
    }

    static getDerivedStateFromProps(props, state) {
        return {
            email: props.email,
            name: props.name,
            serviceAction: props.serviceAction,
            action: props.action,
            actionInfo: props.actionInfo,
            serviceReaction: props.serviceReaction,
            reaction: props.reaction,
            reactionInfo: props.reactionInfo
        };
    }

    render() {
        return (
            <p> 
                {this.state.name}
                <br></br>
                {this.state.serviceAction}
                <br></br>
                {this.state.action}
                <br></br>
                {this.state.actionInfo}
                <br></br>
                {this.state.serviceReaction}
                <br></br>
                {this.state.reaction}
                <br></br>
                {this.state.reactionInfo}
            </p>
        )
    }
}

export default Cards;